test/test_blocks.py

```diff
@@ -463,6 +466,115 @@ def test_raise_error_if_event_queued_but_queue_not_enabled(self):
         demo.close()
 
 
+class TestTempFile:
+    def test_pil_images_hashed(self, tmp_path, connect):
+        def create_images(n_images):
+            a = Image.new("RGB", (512, 512), "red")
+            b = Image.new("RGB", (512, 512), "green")
+            c = Image.new("RGB", (512, 512), "blue")
+
+            res = [a, b, c][:n_images]
+            random.shuffle(res)
+            return res
+
+        with mock.patch.dict(os.environ, {"GRADIO_TEMP_DIR": str(tmp_path)}):
```

Comment:
Repeated for all of these tests:

Py.test has a `monkeypatch` fixture with `setenv` (which obviously resets the environment back when the fixture dissolves), so you can get by with one less nesting level and shorter tests with

```python
monkeypatch.setenv("GRADIO_TEMP_DIR", str(temp_path))
```
instead of `mock.patch.dict`.

Reply:
Nice! Thanks